src/main/java/org/folio/consortia/service/impl/TenantServiceImpl.java

```diff
@@ -149,4 +172,76 @@ private ConsortiaConfiguration createConsortiaConfigurationBody(String tenantId)
     configuration.setCentralTenantId(tenantId);
     return configuration;
   }
+
+  private void createShadowAdminUserWithPermissions(User user) {
+    User userOptional = userTenantService.getUser(UUID.fromString(user.getId()));
+    if (Objects.isNull(userOptional.getId())) {
+      userOptional = createUser(user);
+    }
+    Optional<PermissionUser> permissionUserOptional = permissionService.getPermissionUserById(userOptional.getId());
+    if (permissionUserOptional.isPresent()) {
+      addPermissions(permissionUserOptional.get());
+    } else {
+      createPermissionUser(user.getId());
+    }
+  }
+
+  private PermissionUser createPermissionUser(String userId) {
+    List<String> perms = readPermissionsFromResource(PERMISSIONS_FILE_PATH);
+
+    if (CollectionUtils.isEmpty(perms)) {
+      throw new IllegalStateException("No user permissions found in " + PERMISSIONS_FILE_PATH);
+    }
+    return permissionService.createPermissionUserWithPermissions(UUID.randomUUID().toString(), userId, perms);
+  }
+
+  private List<String> readPermissionsFromResource(String permissionsFilePath) {
+    List<String> result = new ArrayList<>();
+    var url = Resources.getResource(permissionsFilePath);
+
+    try {
+      result = Resources.readLines(url, StandardCharsets.UTF_8);
+    } catch (IOException e) {
+      log.error("Can't read user permissions from %s.", permissionsFilePath, e);
+    }
+
+    return result;
+  }
+
+  private void addPermissions(PermissionUser permissionUser) {
+    var permissions = readPermissionsFromResource(PERMISSIONS_FILE_PATH);
+    if (CollectionUtils.isEmpty(permissions)) {
+      throw new IllegalStateException("No user permissions found in " + PERMISSIONS_FILE_PATH);
+    }
+
+    permissions.removeAll(permissionUser.getPermissions());
+    permissions.forEach(permission -> {
+      var p = new Permission();
+      p.setPermissionName(permission);
+      try {
+        log.info("Adding to user {} permission {}.", permissionUser.getUserId(), p);
+        permissionService.addPermissionToUser(permissionUser.getUserId(), p);
+      } catch (Exception e) {
+        log.error("Error adding permission %s to %s.", permission, permissionUser.getUserId(), e);
+      }
+    });
+  }
+
+  private User createUser(User user) {
+    log.info("Creating {}.", user);
```

Comment:
lets replace to userId, we could log full user object because it can contain sensitive info